src/shared_modules/content_manager/src/components/offlineDownloader.hpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Wazuh content manager
+ * Copyright (C) 2015, Wazuh Inc.
+ * October 24, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _OFFLINE_DOWNLOADER_HPP
+#define _OFFLINE_DOWNLOADER_HPP
+
+#include "hashHelper.h"
+#include "json.hpp"
+#include "stringHelper.h"
+#include "updaterContext.hpp"
+#include "utils/chainOfResponsability.hpp"
+#include <array>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <utility>
+
+/**
+ * @class OfflineDownloader
+ *
+ * @brief Class in charge of copying a file from the filesystem and update the context accordingly, as a step of a chain
+ * of responsibility.
+ *
+ */
+class OfflineDownloader final : public AbstractHandler<std::shared_ptr<UpdaterContext>>
+{
+private:
+    /**
+     * @brief Pushes the state of the current stage into the data field of the context.
+     *
+     * @param contextData Reference to the context data.
+     * @param status Status to be pushed.
+     */
+    void pushStageStatus(nlohmann::json& contextData, std::string status) const
+    {
+        auto statusObject = nlohmann::json::object();
+        statusObject["stage"] = "OfflineDownloader";
+        statusObject["status"] = std::move(status);
+
+        contextData.at("stageStatus").push_back(std::move(statusObject));
+    }
+
+    /**
+     * @brief Function to calculate the hash of a file.
+     *
+     * @param filepath Path to the file.
+     * @return std::string Digest vector.
+     */
+    std::string hashFile(const std::filesystem::path& filepath) const
```

Comment:
Question, why not add this function to `utils/hashHelper.h`?

Reply:
Well, we should. There are other locations where this helper function is also repeated:
- https://github.com/wazuh/wazuh/blob/e766d074c057d718c3956f65e15adf3fe5597944/src/shared_modules/utils/tests/xzHelper_test.cpp#L26
- https://github.com/wazuh/wazuh/blob/e766d074c057d718c3956f65e15adf3fe5597944/src/shared_modules/utils/tests/zlibHelper_test.cpp#L24

Anyway, that would require testing and stuff. What do you think @Damian-Mangold ?

Reply:
Let's leave it like this for now. I make a note to do a refactoring in the future, we should even evaluate using the intelligence-platform `hashHelper.h`.